### ISSUE TITLE
Please contact support@nylas.com. An unforeseen error has occurred.

### ISSUE THREAD

**Issue by galindro** [nylas/nylas-mail#3371]
##### Are there any related issues?
Maybe this: https://github.com/nylas/nylas-mail/issues/3211

##### What operating system are you using?
```bash
Distributor ID:	Debian
Description:	Debian GNU/Linux 8.6 (jessie)
Release:	8.6
Codename:	jessie
```

##### What version of Nylas Mail are you using?
```bash
ii  nylas-mail                            2.0.10                               amd64        The best email app for people and teams at work
```

##### Do you have any third-party plugins installed? If so, which ones?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
No

##### Is the issue reproducible with a particular attachment, message, signature, etc?
No

I'm trying to add my private IMAP account  from the [Federal University of Santa Catarina - Brazil](http://ufsc.br)  into Nylas but this error was occurred:

![image](https://cloud.githubusercontent.com/assets/4622097/24956101/772a0590-1f5c-11e7-86a2-e246f59bf4a7.png)

This was the error showed in terminal:
```bash
$ nylas-mail 
Gtk-Message: Failed to load module "canberra-gtk-module"
Streaming log data to /tmp/Nylas-Mail-31111.log
App load time: 527ms
{ Error
    at new APIError (/src/flux/errors.es6:57:19)
    at Request._callback (/src/flux/nylas-api-request.es6:75:26)
    at Request.self.callback (/usr/share/nylas-mail/resources/app.asar/node_modules/request/request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/usr/share/nylas-mail/resources/app.asar/node_modules/request/request.js:1081:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/usr/share/nylas-mail/resources/app.asar/node_modules/request/request.js:1001:12)
    at IncomingMessage.g (events.js:286:16)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
  name: 'APIError',
  error: null,
  body: 
   { message: 'Please contact support@nylas.com. An unforeseen error has occurred.',
     type: 'api_error' },
  requestOptions: 
   { url: 'https://n1.nylas.com/auth',
     method: 'POST',
     json: true,
     timeout: 180000,
     path: '/auth',
     body: 
      { provider: 'imap',
        email: 'bruno.galindro@ufsc.br',
        name: 'Bruno Galindro da Costa',
        settings: [Object] } },
  statusCode: 500,
  message: 'Please contact support@nylas.com. An unforeseen error has occurred.',
  errorType: 'api_error' } { pluginIds: [] }
```

UFSC Mail service is used by all university's community and doesn't have any network restrictions to access it (like VPN, etc...).

**Comment by andresgabrielrc** [comment]
Same error for me
Manjaro Linux
Kernel 4.9.22


`{ Error: Error: Error in IMAP command received by server.
    at NylasError (errors.es6:1:8)
    at IMAPProtocolError (imap-errors.es6:22:5)
    at convertImapError (imap-errors.es6:140:15)
    at wrappedReject (imap-connection.es6:245:34)
    at imap-connection.es6:264:11
    at throw (<anonymous>)
    at step (imap-connection.es6:9:1)
    at imap-connection.es6:9:1
    at tryCatcher (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/promise.js:687:18)
    at Async._drainQueue (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/async.js:143:10)
    at Async.drainQueues (/usr/share/nylas-mail/resources/app.asar/node_modules/bluebird/js/release/async.js:17:14)
  message: 'Error: Error in IMAP command received by server.',
  userMessage: 'IMAP protocol error. Please contact support@nylas.com.',
  statusCode: 401,
  source: 'protocol' } { pluginIds: [] }
Creation of this event was denied due to rate limiting
429
`

**Comment by jstejada** [comment]
thanks for the report! we are looking into this one

**Comment by jonathanlaf** [comment]
have you tried ticking Allow unsecure SSL ? That was the problem for me, since my EXIM server do not have a matching SMTP banner it was considered unsecure SSL.

**Comment by galindro** [comment]
@jonathanlaf what did you mean about insecure SSL? 

This client needs to just work with my dovecot and postfix configuration as many other clients that just works (thunderbird, outlook, icedove, etc...)

**Comment by jstejada** [comment]
Thanks for your patience! We’ve just released version 2.0.18 which should allow you to authenticate despite the error your are seeing. This error is likely caused by the presence of an antivirus in your system, so if you disable your anti-virus for Nylas Mail, you should stop seeing this error when you authenticate.

Please let me know if you continue having issues. Thanks again for your patience!
